### PR TITLE
Retreats

### PR DESCRIPTION
**Pull request by jasonekstrom**
Dislodge logic added

### CODE REVIEW

File: beta-src/src/state/game/game-api-slice.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ const resetOrder = (state) => {
   } = current(state);
   if (type !== "hold") {
     const command: GameCommand = {
-      command: "NONE",
+      command: type === "Retreat" ? "DISLODGED" : "NONE",
```
Comment (bryancodazen) [inline beta-src/src/state/game/game-api-slice.ts:153]: type here should be an `OrderType` which is in `/src/types/state/OrderType`. We should keep them all lowercase for consistency.

File: beta-src/src/state/game/game-api-slice.ts [hunk 2]
```diff
@@ -178,7 +178,11 @@ const startNewOrder = (
   state.order.orderID = orderForUnit.id;
   state.order.onTerritory = onTerritory;
   state.order.toTerritory = null;
-  delete state.order.type;
+  if (orderForUnit.type === "Retreat") {
+    state.order.type = "Retreat";
```
Comment (bryancodazen) [inline beta-src/src/state/game/game-api-slice.ts:182]: type should not have a value until an order is completed. Please look at how the hold/move orders manipulate the `order` property in the global state to see how it should work.

File: beta-src/src/state/game/game-api-slice.ts [hunk 3]
```diff
@@ -293,13 +297,41 @@ const drawBuilds = (state) => {
...
+  if (phase === "Retreats") {
+    currentOrders.forEach((order, index) => {
+      const command: GameCommand = {
+        command: "DISLODGED",
+      };
+      setCommand(state, command, "unitCommands", order.unitID);
+    });
+  }
+};
+
```
Comment (bryancodazen) [inline beta-src/src/state/game/game-api-slice.ts:326]: In game API slice you are calling this function anytime the `ordersMeta` property is updated. This means if you update an order to a disband icon this code will change the icon back to dislodged. 
Screen Shot 2022-04-26 at 3 23 12 PM

**Review by bryancodazen — CHANGES_REQUESTED** [review]
Don't delete package-lock.json in the PR. 

**In game-api-slice.ts:**

Remove unused variables and `command` in the code below is showing an error:
Screen Shot 2022-04-26 at 3 26 14 PM

Screen Shot 2022-04-26 at 3 27 09 PM

Screen Shot 2022-04-26 at 3 28 02 PM

File: beta-src/src/state/game/game-api-slice.ts (hunk 3 above)
Comment (sariacodazen) [reply beta-src/src/state/game/game-api-slice.ts:326]: @bryancodazen correct me if I am wrong but I believe @jasonekstrom you need to check to make sure there are no ready orders in game status, or saved orders in ordersMeta (checking its saved property) pertaining to the unit in order before setting its ui state to dislodge. This is because whenever you process either a move or hold order through clicking a terr in retreat, it would remove dislodge state, and the user would then save it which would break the condition mentioned above. If they refresh before save, icon goes back to dislodge state because order was never saved.

**Comment by jasonekstrom** [comment]
I addressed all these with my new commit

**Comment by bryancodazen** [comment]
Hey @jasonekstrom , please merge staging and resolve conflicts.

**Review by mariomelchor — CHANGES_REQUESTED** [review]

File: beta-src/src/state/game/game-api-slice.ts (hunk 3 above)
Comment (jasonekstrom) [reply beta-src/src/state/game/game-api-slice.ts]: Updated in my newest commit

File: beta-src/src/components/ui/WDMoveControls.tsx [hunk 4]
```diff
@@ -87,6 +87,7 @@ const WDMoveControls: React.FC<WDMoveControlsProps> = function ({
             ? { notready: "on" }
             : { ready: "on" };
         }
+
```
Comment (bryancodazen) [inline beta-src/src/components/ui/WDMoveControls.tsx:90]: remove unnecessary space

File: beta-src/src/state/game/game-api-slice.ts [hunk 5]
```diff
@@ -310,6 +336,7 @@ const updateOrdersMeta = (state, updates: EditOrderMeta) => {
     };
   });
   drawOrders(state);
+  updateDislodged(state);
```
Comment (bryancodazen) [inline beta-src/src/state/game/game-api-slice.ts:339]: remove `updateDislodged` from here and place it in drawOrders

File: beta-src/src/utils/map/getOrdersMeta.ts [hunk 6]
```diff
@@ -62,6 +75,20 @@ export default function getOrdersMeta(data, phase): Props {
...
+            allowedBorderCrossings = allowedBorderCrossings.filter(
+              (crossing) => !crossing.unitID && !crossing.standoff,
+            );
+            if (occupiedTerritory) {
+              allowedBorderCrossings = allowedBorderCrossings.filter(
+                (crossing) =>
+                  crossing.id !== occupiedTerritory.occupiedFromTerrID,
+              );
+            }
+          }
```
Comment (bryancodazen) [inline beta-src/src/utils/map/getOrdersMeta.ts:91]: it appears we are re-assigning allowedBorderCrossings too much, we should move the occupiedTerritory check into the filter above if possible.

File: beta-src/src/utils/map/getOrdersMeta.ts [hunk 7]
```diff
@@ -35,15 +36,27 @@ export default function getOrdersMeta(data, phase): Props {
...
+          orderUnit = new UnitClass({
+            ...units[unitID],
+            Territory: territories[units[unitID].terrID],
+          });
+        }
+
         if (orderUnit) {
-          newOrders.push(new OrderClass(newBoard, o, orderUnit));
+          const tempOrder = new OrderClass(newBoard, o, orderUnit);
+          newOrders.push(tempOrder);
```
Comment (bryancodazen) [inline beta-src/src/utils/map/getOrdersMeta.ts:50]: why is this necessary to place into its own variable?

**Review by bryancodazen — CHANGES_REQUESTED** [review]
